File: frontend/classes/Phase.ts
interface PhaseData {
  id: number
  name: string
  index: number
  turn: number
}

class Phase {
  id: number
  name: string
  index: number
  turn: number

  constructor(data: PhaseData) {
    this.id = data.id
    this.name = data.name
    this.index = data.index
    this.turn = data.turn
  }
}

export default Phase

File: backend/rorapp/models/phase.py
from django.db import models
from rorapp.models.turn import Turn


# Model for representing phases
class Phase(models.Model):
    name = models.CharField(max_length=10)
    index = models.PositiveIntegerField()
    turn = models.ForeignKey(Turn, on_delete=models.CASCADE)
